include/numerics/sparse_matrix.h

```diff
@@ -583,6 +583,32 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T>>,
    */
   virtual void scale(const T scale);
 
+  /**
+   * @returns Whether the matrix supports hash table assembly
+   */
+  virtual bool supports_hash_table() const { return false; }
+
+  /**
+   * @returns Whether a preference for hash table vs. preallocation has been set
+   */
+  bool assembly_preference_set() const { return _assembly_preference_set; }
```

Comment:
How does this differ from just `use_hash_table()`?  If `use_hash_table()` returns `true`, doesn't that mean I've set that as a preference?  If it's `false`, doesn't that in general mean that I've left the default `_use_hash_table` setting on rather than that I've specifically called `do_what_you_were_already_going_to()`?